lib/task-data/schemas/download-file.json

```diff
@@ -0,0 +1,60 @@
+{
+    "$schema": "rackhd-task-schema.json",
+    "copyright": "Copyright 2016, EMC, Inc.",
+    "title": "Firmware Update",
+    "description": "The schema for firmware update task",
+    "describeJob": "Job.Linux.Commands",
+    "definitions": {
+        "File": {
+            "description": "The filename to download",
+            "type": "string",
+            "minLength": 1
+        },
+        "DownloadDir": {
+            "description": "The microkernel directory where the file will be downloaded to",
+            "type": "string",
+            "minLength": 1
+        },
+        "FileMd5Uri": {
+            "description": "The URL that returns the MD5 of the target file",
+            "type": "string",
+            "minLength": 1,
+            "pattern": "^https?://"
+        },
+        "FileUri": {
+            "description": "The URL that is used to download the target file",
+            "type": "string",
+            "minLength": 1,
+            "pattern": "^https?://"
```

Comment:
Actually, I tested the `abc/` is still a valid uri format, but this is not what we want. For a format uri is not enough here.